omniqueue/src/backends/azure_queue_storage.rs

```diff
@@ -0,0 +1,218 @@
+                config: config.clone(),
+            },
+            AqsConsumer {
+                client: client.clone(),
+                config: config.clone(),
+            },
+        ))
+    }
+
+    async fn producing_half(config: Self::Config) -> Result<AqsProducer> {
+        let client = get_client(&config);
+        Ok(AqsProducer { client, config })
+    }
+
+    async fn consuming_half(config: Self::Config) -> Result<AqsConsumer> {
+        let client = get_client(&config);
+        Ok(AqsConsumer { client, config })
+    }
+}
+
+pub struct AqsProducer {
+    client: QueueClient,
+    config: AqsConfig,
+}
+
+impl AqsProducer {
+    pub async fn send_raw(&self, payload: &str) -> Result<()> {
+        self.send_raw_scheduled(payload, Duration::ZERO).await
+    }
+
+    pub async fn send_raw_scheduled(&self, payload: &str, delay: Duration) -> Result<()> {
+        self.client
+            .put_message(payload)
+            .visibility_timeout(delay)
+            .ttl(self.config.message_ttl)
+            .await
+            .map_err(QueueError::generic)
+            .map(|_| ())
+    }
+
+    pub async fn send_serde_json<P: Serialize + Sync>(&self, payload: &P) -> Result<()> {
+        let payload = serde_json::to_string(payload)?;
+        self.send_raw(&payload).await
+    }
+    pub async fn send_serde_json_scheduled<P: Serialize + Sync>(
+        &self,
+        payload: &P,
+        delay: Duration,
+    ) -> Result<()> {
+        let payload = serde_json::to_string(payload)?;
+        self.send_raw_scheduled(&payload, delay).await
+    }
+}
+
+impl_queue_producer!(AqsProducer, String);
+impl_scheduled_queue_producer!(AqsProducer, String);
+
+pub struct AqsConsumer {
+    client: QueueClient,
+    config: AqsConfig,
+}
+
+struct AqsAcker {
+    client: QueueClient,
+    already_acked_or_nacked: bool,
+    pop_receipt: PopReceipt,
+}
+
+#[async_trait]
+impl Acker for AqsAcker {
+    async fn ack(&mut self) -> Result<()> {
+        if self.already_acked_or_nacked {
+            return Err(QueueError::CannotAckOrNackTwice);
+        }
+        self.already_acked_or_nacked = true;
+        self.client
+            .pop_receipt_client(self.pop_receipt.clone())
+            .delete()
+            .await
+            .map_err(QueueError::generic)
+            .map(|_| ())
+    }
+
+    async fn nack(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl AqsConsumer {
+    fn wrap_message(&self, message: &Message) -> Delivery {
+        Delivery {
+            acker: Box::new(AqsAcker {
+                client: self.client.clone(),
+                pop_receipt: message.pop_receipt(),
+                already_acked_or_nacked: false,
+            }),
+            payload: Some(message.message_text.as_bytes().to_owned()),
+        }
+    }
+
+    /// Note that blocking receives are not supported by Azure Queue Storage
+    /// and will return immediately if no messages are available for delivery
+    /// in the queue.
```

Comment:
Can you copy-paste this to the documentation of `AqsBackend` and/or the module docs, as well as the `AqsConsumer` type, for extra visibility?